File: DSA/RATNA_KHANNA_590013141/DAY_66_QUESTION_2.c
#include <stdio.h>

int main() {
    int n;
    printf("Enter number of days: ");
    scanf("%d", &n);

    int prices[n];
    printf("Enter stock prices:\n");
    for(int i=0;i<n;i++) scanf("%d",&prices[i]);

    int min_price = prices[0], max_profit = 0;
    for(int i=1;i<n;i++) {
        if(prices[i] - min_price > max_profit)
            max_profit = prices[i] - min_price;
        if(prices[i] < min_price)
            min_price = prices[i];
    }

    printf("Maximum profit with one transaction: %d\n", max_profit);
    return 0;
}

File: DSA/AnshulJain_590014934/Day_66_Question_2.py
n = int(input("Enter number of days: "))# taking input from the user 
prices = list(map(int, input("Enter stock prices: ").split()))#taking input from the user
min_price = prices[0]
max_profit = 0
for price in prices:
    min_price = min(min_price, price)
    max_profit = max(max_profit, price - min_price)
print(max_profit)
#for exapmple:
# Input:
# Enter number of days: 5
# Enter stock prices: 7 1 5 3 6 4
# Output:5